tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/viewers/TreeViewerTest.java

```diff
@@ -50,4 +56,25 @@ protected String getItemText(int at) {
 		return tree.getItems()[at].getText();
 	}
 
+	@Test
+	public void testAutoExpandTrivialPathThroughEvent() {
+		TestElement modelRoot = TestElement.createModel(5, 1);
+		TestElement trivialPathRoot = modelRoot.getFirstChild();
+		fViewer.setInput(modelRoot);
+
+		System.out.println("TreeViewerTest - 0: " + trivialPathRoot);
+		fTreeViewer.setAutoExpandOnSingleChildLevels(2);
+		Event event = new Event();
+		System.out.println("TreeViewerTest - 1: " + event.item);
+		event.item = fViewer.testFindItem(trivialPathRoot);
```

Comment:
Can you please try replacing this line as follows:
```suggestion
		event.item = ((Tree) fViewer.getControl()).getItem(0);
```

Reply:
Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/fb256ed9423ce9f552aa278fa3893d833ccc0e63